ninshiki_yolo/node/detector.py

```diff
@@ -86,40 +88,38 @@ def listener_callback(self, message: MsgType):
 
         # Clear message list
         self.detection_result.detected_objects.clear()
+    
+    def define_model(self, image: np.ndarray):
+        with open(self.file_name, 'rt') as f:
+            self.classes = f.read().rstrip('\n').split('\n')
+        
+        self.net.getLayerNames()
+        layerOutput = self.net.getUnconnectedOutLayersNames()
 
-    def detection(self, image: np.ndarray, detection_result: MsgType):
-        class_file = self.names
-        classes = None
-        with open(class_file, 'rt') as f:
-            classes = f.read().rstrip('\n').split('\n')
-        model_configuration = self.config
-        model_weights = self.weights
-
-        net = cv2.dnn.readNetFromDarknet(model_configuration, model_weights)
-        net.getLayerNames()
-        layerOutput = net.getUnconnectedOutLayersNames()
-
-        net.setPreferableBackend(cv2.dnn.DNN_BACKEND_OPENCV)
-        net.setPreferableTarget(cv2.dnn.DNN_TARGET_CPU)
+        self.net.setPreferableBackend(cv2.dnn.DNN_BACKEND_OPENCV)
+        self.net.setPreferableTarget(cv2.dnn.DNN_TARGET_CPU)
```

Comment:
this should be more flexible to set

Reply:
Should I change the name of define_model function? Because I think it is not suitable for its functionalities. Maybe like pass_image_to_network or do you have another more suitable name?